.github/workflows/dependabot-uv-lock.yml

```diff
@@ -0,0 +1,34 @@
+name: uv
+
+on:
+  pull_request:
+    paths:
+      - "pyproject.toml"
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  lock:
+    runs-on: ubuntu-latest
+    env:
+      GH_PAT: ${{ secrets.GH_PAT }}
+    steps:
+      - name: Checkout with token
+        uses: actions/checkout@v4
+        if: ${{ env.GH_PAT != '' }}
+        with:
+          token: ${{ env.GH_PAT }}
+
+      - name: Checkout without token
+        uses: actions/checkout@v4
+        if: ${{ env.GH_PAT == '' }}
```

Comment:
Added this fallback in case the secret isn't accessible, which unfortunately the case sometimes...